packages/emptystate/src/react/use-resize-observer.js

```diff
@@ -0,0 +1,50 @@
+import { useEffect, useRef, useState } from 'react'
+import ResizeObserver from 'resize-observer-polyfill'
+
+const isBrowser = typeof window !== 'undefined'
+
+export default function useResizeObserver() {
+  const ref = useRef()
+
+  const [width, setWidth] = useState(0)
+  const [height, setHeight] = useState(0)
+
+  useEffect(() => {
+    let subscribed = true
+    const { current: el } = ref
+
+    function handleResize(entries) {
+      if (!subscribed || !isBrowser) return
+      const { contentRect } = entries[0]
+
+      setWidth(contentRect.width)
+      setHeight(contentRect.height)
+    }
+
+    const debouncedResize = debounce(150, handleResize)
+    const observer = new ResizeObserver(debouncedResize)
```

Comment:
`useEffect` is potentially called multiple times.  Does creating new instance here create a problem?  Do we want to ever just have one?

Reply:
because the second arg to useEffect is an empty array it should(🤞) just run once on mount. i'm still fuzzy on the api so i'll verify that's how it works